racket/collects/racket/private/prop-template-metafunction.rkt

```diff
@@ -0,0 +1,88 @@
+(module prop-template-metafunction '#%kernel
```

Comment:
I would prefer this whole module's contents to be put in racket/private/template.rkt instead of a separate file. And that means this code can use the "small-scheme.rkt" language, including `define` and `cond` and so on.